shell/platform/android/platform_view_android.cc

```diff
@@ -46,17 +48,17 @@ std::unique_ptr<AndroidSurface> AndroidSurfaceFactoryImpl::CreateSurface() {
     case AndroidRenderingAPI::kSoftware:
       return std::make_unique<AndroidSurfaceSoftware>(android_context_,
                                                       jni_facade_);
-    case AndroidRenderingAPI::kOpenGLES:
+    case AndroidRenderingAPI::kGPU:
       if (enable_impeller_) {
-// TODO(kaushikiska@): Enable this after wiring a preference for Vulkan backend.
-#if false
-        return std::make_unique<AndroidSurfaceVulkanImpeller>(
-            android_context_, jni_facade_, enable_vulkan_validation_);
-
-#else
+        if (!impeller_use_gl_) {
+          auto vk_surface = std::make_unique<AndroidSurfaceVulkanImpeller>(
+              android_context_, jni_facade_, enable_vulkan_validation_);
+          if (vk_surface->IsValid()) {
+            return vk_surface;
+          }
+        }
```

Comment:
An info log here saying we are falling back to GL would be good.

Reply:
There's a log later saying we're using ht eGL backend in this path.